testpset1.cpp

```diff
@@ -5,5 +5,22 @@ using namespace std;
 int main()
 {
     init_kalloc();
+    void* ptr = kalloc(1UL<<21);
+    kalloc(1UL<<13);
+    kfree(ptr);
+    void* ptr1 = kalloc(1UL<<13);
 
-}
\ No newline at end of file
+    void* ptr2 = kalloc(1UL<<13);
+    kalloc(1UL<<15);
+    kfree(ptr1);
+    kfree(ptr2);
+    kalloc(1UL<<13);
```

Comment:
I think you could do a lot more to test your code (or maybe you had more tests and got rid of them), but it works!